File: js/Number_of_Segments_in_a_String.js
/**
 * @param {string} s
 * @return {number}
 */
var countSegments = function(s) {
    s = s.trim();
    if (s === '') {
        return 0;
    }
    let arr = s.split(' ');
    let res = 0;
    arr.forEach(item => {
        if (item.trim() !== '') {
            res ++;
        }
    });
    return res;
};

File: go/Number_of_Segments_in_a_String.go
package main

import "fmt"
import "strings"

func countSegments(s string) int {
    s = strings.TrimSpace(s)
    if s == "" {
        return 0
    }
    arr := strings.Split(s, " ")
    res := 0
    for _, item := range arr {
        if strings.TrimSpace(item) != "" {
            res ++
        }
    }
    return res
}

func main() {
    fmt.Println(countSegments(", , , ,        a, eaefa"))
}
